lib/core/src/sdk.rs

```diff
@@ -3571,6 +3585,138 @@ impl LiquidSdk {
     }
 }
 
+/// Uses existing `ln_address` if it's available, otherwise tries to extract `ln_address` from `metadata_str`
+fn derive_ln_address(
+    ln_address: Option<String>,
+    metadata_str: &str,
+    domain: &str,
+) -> Option<String> {
+    match ln_address {
+        Some(addr) => {
+            info!("Using existing LN address: {}", addr);
+            Some(addr)
+        }
+        None => match extract_ln_address_from_metadata(metadata_str, domain) {
+            Some(addr) => {
+                info!("Extracted LN address from metadata_str: {}", addr);
+                Some(addr)
+            }
+            None => {
+                warn!(
+                    "No LN address found in metadata_str for domain '{}'",
+                    domain
+                );
+                None
+            }
+        },
+    }
+}
+
+/// Extracts `ln_address` from `metadata_str` and validates it against given `domain`
+fn extract_ln_address_from_metadata(metadata_str: &str, expected_domain: &str) -> Option<String> {
+    if metadata_str.is_empty() {
+        warn!("Metadata string is empty, no LN address available.");
```

Comment:
We already log above when none is found

Reply:
@dangeross Thanks for the review! Most of the logs will be removed later on.

I'm aware of the redundancies, they were helpful for tracking the code without needing to rebuild. I'm keeping them for debugging purposes for now. Redundant ones will be removed when PR is marked for review.